### PR TITLE
Added option to parse hyperblock with txs finalized at source

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
Added the `?notarizedAtSource=true` option so the resulting hyperblock will only contain transactions that have been notarized at source, as opposed to the original hyperblock that only returned transactions notarized at destination

**Testing procedure**:
- on a testnet use this proxy branch
- find a metablock that has notarized shard headers
- query `<proxy-url>:<proxy-port>/hyperblock/by-nonce/<nonce from previous step>?notarizedAtSource=true` and you should only see transactions that are not mandatory final (for cross-check, you can see that one of the transactions was only executed on source, while the destination execution was at a later time)
- query without the `?notarizedAtSource=true` and/or with `?notarizedAtSource=false` and only finalized transaction should be seen

### CODE REVIEW

File: common/options.go [hunk 1]
```diff
@@ -10,6 +10,8 @@ const (
 	UrlParameterWithTransactions = "withTxs"
 	// UrlParameterWithLogs represents the name of an URL parameter
 	UrlParameterWithLogs = "withLogs"
+	// UrlParameterNotarizedAtSource represents the name of an URL parameter
+	UrlParameterNotarizedAtSource = "notarized-at-source"
```
Comment (andreibancioiu) [inline common/options.go:14]: The other options follow the `camelCase` format. What should we do in this case?

File: process/hyperblockBuilder_test.go [hunk 2]
```diff
@@ -61,3 +61,56 @@ func TestHyperblockBuilder(t *testing.T) {
 		{Sender: "carol", Receiver: "carol"},
 	}, hyperblock.Transactions)
 }
+
+func TestHyperblockBuilderWithNotarizedAtSourceTxs(t *testing.T) {
```
Comment (andreibancioiu) [inline process/hyperblockBuilder_test.go:65]: :+1:

File: common/options.go (hunk 1 above)
Comment (bogdan-rosianu) [reply common/options.go]: renamed to be camelCase

**Review by ccorcoveanu — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by gabi-vuls — COMMENTED** [review]
system test passed

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**